.github/workflows/reusable_publish_version.yml

```diff
@@ -38,12 +37,121 @@ jobs:
       git-ref: ${{ steps.outputs.outputs.git-ref }}
       name: ${{ steps.outputs.outputs.name }}
     steps:
+      - name: Install stable toolchain
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: stable
+
+      - name: Checkout sources
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.git-ref || github.ref_name }}
+
+      - name: Install a TOML parser
+        run: cargo install --force --locked --version 0.8.1 taplo-cli
+
+      - name: Configure git
+        run: |
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config user.name "github-actions[bot]"
+          git config --add --bool push.autoSetupRemote true
+
+      - name: Patch release version
+        if: ${{ inputs.environment == 'release' }}
+        run: |
+          set -x
+
+          currentVersion=$(/home/runner/.cargo/bin/taplo get -f lib/Cargo.toml "package.version")
+
+          if [[ $currentVersion == *"-beta"* ]]; then
+            major=$(echo $currentVersion | tr "." "\n" | sed -n 1p)
+            minor=$(echo $currentVersion | tr "." "\n" | sed -n 2p)
+            betaNum=$(echo $currentVersion | tr "." "\n" | sed -n 4p)
+            version=${major}.${minor}.$(($betaNum - 1))
```

Comment:
do we want to decrease the beta number? 🤔 could  you provide an example?

Reply:
Yes we need to do this. Binaries and tags use the format `x.y.0-beta.z` which maps to `x.y.z-1` for the crate version. For example:

- `1.1.0-beta.1` maps to `1.1.0`
- `1.1.0-beta.2` maps to `1.1.1`
- `1.1.0-beta.3` maps to `1.1.2`

Reply:
ah understood 👍